src/compile/layout.ts

```diff
@@ -86,7 +86,7 @@ function parseUnitSizeLayout(model: UnitModel, channel: Channel): SizeComponent
 }
 
 function unitSizeExpr(model: UnitModel, channel: Channel, nonOrdinalSize: number): string {
-  if (model.has(channel)) {
+  if (model.has(channel) || (channel === X && model.has(X2)) || (channel === Y && model.has(Y2))) {
```

Comment:
If I understand correctly, even if `model.has(X)` is `false`, we would have initialized scale `x`.  

Thus, can you try if replacing the predicate inside `if` with  `model.scale(channel) !== null` working?  

(If it works, we can eliminate `model.has()` as well. )